test/gtest/common/test_helpers.h

```diff
@@ -384,17 +384,35 @@ class message_stream {
     ~message_stream();
 
     template <typename T>
-    std::ostream& operator<<(const T& value) const {
-        return std::cout << value;
+    message_stream& operator<<(const T& value) {
+        msg << value;
+        return *this;
+    }
+
+    message_stream& operator<< (std::ostream&(*f)(std::ostream&)) {
+        if (f == (std::basic_ostream<char>& (*)(std::basic_ostream<char>&)) &std::flush) {
+            std::string s = msg.str();
+            if (!s.empty()) {
+                std::cout << s << std::flush;
+            }
+            msg.str("");
```

Comment:
Double clear?

Reply:
No, because msg.clear() clears only error state bits and I also need to reset the string content.

Reply:
I see msg is not a std::string.. btw, you can move it under empty check above.